applications/Preprocessor/lib/elementShape_impl.h

```diff
@@ -163,34 +74,43 @@ std::enable_if_t<(entityDimension >= 0 && entityDimension < dimension &&
                         "%, but you asked for shape %",
                         getNumberOfEntities<entityDimension>(), dimension,
                         entityIndex);
-    return shapeData.template getAdjacentShapes<
-        entityDimension>()[targetDimension][entityIndex];
+    return shapeParts.template get<entityDimension>()[entityIndex]
+        .getAdjacentShapes(targetDimension);
 }
 
 template <std::size_t dimension>
-template <std::size_t d, std::size_t shapeDimension>
+template <int d, std::size_t shapeDimension>
 bool ElementShape<dimension>::checkSingleShape(
     const ElementShape<shapeDimension>* boundingShape,
     std::vector<std::size_t> boundaryNodes, std::size_t currentIndex,
```

Comment:
we need the copy here?

Reply:
out of scope of the PR

Reply:
What copy do you mean?

Reply:
boundaryNotes. Sorry.

Reply:
Does not seem like it, but as outside and it is only called like a few times (no time usage) I don't it is worth fixing it now.